webbpsf/roman.py

```diff
@@ -808,12 +800,10 @@ def lock_pupil(self, pupil_path):
 
     def unlock_pupil(self):
         """
-        Undoes the effects of lock_pupil() and resets the class to its
-        default state of updating the pupil whenever a detector or
-        filter is changed.
-
-        # UNRESOLVED: how do we know which pupil to reset to? make it clear in the docstring that you're unlocking *and* resetting to default.
-        # UNRESOLVED: if you unlock_pupil_mask, lock_pupil, unlock_pupil, does the pupil mask also reset?
+        Undoes the effects of lock_pupil() by resetting the class to
```

Comment:
This is out of scope for this PR, but at some point it might be worth revisiting some of the complexity that's been added into the WFI classes, such as this lock and unlock pupil functionality. What is the use case for that? Is it still needed? Does anyone actually use this part of the code? I think it may be the case that some of the code added over time is no longer particularly useful, and could potentially be simplified out.

Reply:
Robel and I spoke with Adric about this and Adric said it's useful in his workflows the Roman ETC (and perhaps Pandeia). Robel also told me that there are some other, more intrepid users who like having this functionality.